src/pages/index.js

```diff
@@ -1,67 +1,263 @@
+// -------------------------------------------------------------------------------------
+const imagePopup = new PopupWithImage("#image-modal");
+imagePopup.setEventListeners();
+
+// -------------------------------------------------------------------------------------
+// API Instance
+// -------------------------------------------------------------------------------------
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "b656ee04-ff76-4b23-b20a-1e4991fc7f99",
+});
+
 // -------------------------------------------------------------------------------------
-//                              UserInfo Instance
+// UserInfo Instance
 // -------------------------------------------------------------------------------------
 const userInfo = new UserInfo({
   nameSelector: ".profile__name",
-  descriptionSelector: ".profile__description",
+  aboutSelector: ".profile__description",
 });
+
 // -------------------------------------------------------------------------------------
-//                                 Popup Instances
+// Inital UserInfo Population
 // -------------------------------------------------------------------------------------
-const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
-  handleFormSubmit: (formData) => {
+api
+  .getProfileData()
+  .then((data) => {
     userInfo.setUserInfo({
-      name: formData["name"],
-      description: formData["description"],
+      name: data.name,
+      about: data.about,
     });
-    profileEditPopup.close();
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile data:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserInfo Popup
+// -------------------------------------------------------------------------------------
+const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
+  handleFormSubmit: (formData) => {
+    profileEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editProfileData(formData)
+      .then((data) => {
+        userInfo.setUserInfo({
+          name: data.name,
+          about: data.about,
+        });
+        profileEditPopup.close();
+      })
+      .catch((error) => {
+        console.log("Error updating Profile:", error);
+      })
+      .finally(() => {
+        profileEditPopup.toggleUploadIndicator(false);
+      });
   },
 });
+
 profileEditPopup.setEventListeners();
 
+// -------------------------------------------------------------------------------------
+// User Avatar Instance
+// -------------------------------------------------------------------------------------
+const userAvatar = new UserAvatar({
+  avatarSelector: ".profile__image",
+});
+
+// -------------------------------------------------------------------------------------
+// Inital User Avatar Population
+// -------------------------------------------------------------------------------------
+api
+  .getUserAvatar()
```

Comment:
`getUserAvatar` is not needed because `getProfileData` has the `avatar` in its response from the server